src/DynamoCoreWpf/ViewModels/Core/DynamoViewModel.cs

```diff
@@ -772,28 +779,84 @@ protected DynamoViewModel(StartConfiguration startConfiguration)
             MLDataPipelineExtension = model.ExtensionManager.Extensions.OfType<DynamoMLDataPipelineExtension>().FirstOrDefault();
         }
 
+        private void TaskScheduler_UnobservedTaskException(object sender, UnobservedTaskExceptionEventArgs e)
+        {
+            try
+            {
+                var crashData = new CrashErrorReportArgs(e.Exception);
```

Comment:
This is a little confusing. Why are you initializing a CrashErrorReport when you're not intending this exception to be a crash?

Reply:
CrashErrorReportArgs has code that can take into account InnerExceptions.
InnerExceptions are pretty common in async code.